### PR TITLE
Add local incr_or_init to remove race conditions

### PR DESCRIPTION
**Pull request by es**
As the title says, adds a local `incr_or_init` to remove potential race conditions.

r: @agentzh

### CODE REVIEW

File: lib/resty/limit/conn.lua [hunk 1]
```diff
@@ -23,6 +23,29 @@ local mt = {
 }
 
 
+local function incr_or_init(key, dict)
```
Comment (agentzh) [inline lib/resty/limit/conn.lua]: I think use the shdict native API is better. There is a pending PR for this:

https://github.com/openresty/lua-nginx-module/pull/579

Comment (es) [inline lib/resty/limit/conn.lua]: Any ETA on when that'll be reviewed/ :ship:'d?

Comment (agentzh) [inline lib/resty/limit/conn.lua]: @es No exact ETA. Soon maybe :)

Comment (es) [inline lib/resty/limit/conn.lua]: Would it make sense to ship this PR and then move to the new shdict API once https://github.com/openresty/lua-nginx-module/pull/579 is 🚢 'd?

Comment (agentzh) [inline lib/resty/limit/conn.lua]: @es Nope. This race condition should be minor. And I don't like using the goto feature in your PR. The Lua 5.2 feature can be disabled in the LuaJIT builds (and which is the default) and we shouldn't rely on it.

Comment (es) [inline lib/resty/limit/conn.lua]: In that case I'll close the PR.

**Pull request closed without merging**